### PR TITLE
New map: Wheelstation

### PR DESCRIPTION
**Pull request by brndd**
Here's something I've been working on for the past few months: a brand new map called Wheelstation.

It's a relatively large map, owing to its design of having lots of space gaps for aesthetics (loosely inspired by Goon's Cogmap). In terms of floor area it shouldn't be much larger than Box though. I'd estimate it plays best with population requirements similar to Defficiency.

### Segmented design

full

The station is divided into 5 sections: Northern, Eastern, Southern, Western and Central. The northern section houses Science and Medbay; the Eastern section houses service sector jobs and the escape shuttle dock; the Southern section houses Engineering; the Western section houses the bridge and Security; and the Central section houses arrivals, dorms, lockers, and that kind of thing.

### Segmented atmospherics

The segmented design isn't just for looks. The station has no fewer than 5 (FIVE) separate atmospherics loops, one for each section. This makes atmos more resilient to the epic "hook distro into waste in some backwater maint" sabotage, *and* lets you selectively plasmaflood just those pesky scientist without ruining the day for your buddies at the bar! (though medbay will suffer collateral damage, being on the same loop)

Oh, and it's all layered, which makes piping clean and simple to fix; you will never accidentally mix distro and waste.

### Lots of space

Like, actual space space. That thing the space station is in. You'll see it a lot. One day, when we get space parallax and starlight, this is going to look amazing. For now it just looks kinda pretty.

### No gimmicks... mostly

We've already got Roidstation and Snowmap. I wanted to make a good old regular space station. I hope I succeeded.

### Some gimmicks though

Instead of luring stationeers onto their shuttle in a remote corner of the station, the vox traders have a storefront literally in the middle of the station. This store front has a nitrogen atmosphere (*what could go wrong*) and is ID locked to the vox (not even the captain can get in). The \~lore explanation\~ is that they're renting it.

Nearby, there's another vacant storefront for enterprising crew members to set up shop in.

Telescience is a cursed thing that should be removed from the game. The one on this map is evidence of this. The aftermath of a rogue AI using it to massacre the crew is yet to be cleaned up. This adds an extra challenge to speedrunning powergamers.

## PR notes

This PR may seem like unatomic, but that's just the nature of making an entirely new map from scratch while (mostly) following best practices. I made separate PRs for *most* of the minor mapping-related code changes I did while mapping, but a couple of them slipped through. #24670 is also included in this PR because this map is the reason that PR exists.

:cl:
 * rscadd: Added a new map: Wheelstation.

### CODE REVIEW

**Comment by Kurfursten** [comment]
>arrivals shuttle doors can open directly into space

:eyes:

File: code/game/machinery/alarm.dm [hunk 1]
```diff
@@ -163,6 +163,21 @@
...
+	nitrogen = list(-1, -1, 0.5, 1)
+	carbon_dioxide = list(-1, -1, 0.5, 1)
+	plasma = list(-1, -1, 0.5, 1)
+	n2o = list(-1, -1, 0.5, 1)
+	other = list(-1, -1, 0.5, 1)
+	pressure = list(-1, -1, ONE_ATMOSPHERE*0.01, ONE_ATMOSPHERE*0.05)
+	temperature = list(-1, -1, -1, -1)
+	target_temperature = T0C+20
+	scrubbers_gases = list("oxygen" = 1, "nitrogen" = 1, "carbon_dioxide" = 1, "plasma" = 1, "n2o" = 0)
+
```
Comment (N3X15) [inline code/game/machinery/alarm.dm]: Why.  Just set up a floor vent set to input.

**Comment by brndd** [comment]
>arrivals shuttle doors can open directly into space

Those doors are bolted.

**Comment by Probe1** [comment]
We have parallax I don't know what you mean.  Anyway good job.  👍

File: code/game/machinery/computer/camera.dm [hunk 2]
```diff
@@ -96,6 +96,7 @@ var/list/obj/machinery/camera/cyborg_cams = list(
 	network = list(CAMERANET_THUNDER)
 	density = 0
 	circuit = null
+	layer = ABOVE_WINDOW_LAYER
```
Comment (N3X15) [inline code/game/machinery/computer/camera.dm]: This seems like its own issue.

File: maps/wheelstation.dm [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	)
+
+	holomap_offset_x = list(0,0,0,86,4,0,0,)
+	holomap_offset_y = list(0,0,0,94,10,0,0,)
+
+	center_x = 226
+	center_y = 254
+
+//Randomized wires on select airlocks
+//fuck greyshirts and fuck powergamers
```
Comment (N3X15) [inline maps/wheelstation.dm]: Yeah this is a wonderful idea that will certainly not lead to controversy and shitposting.

**Comment by Probe1** [comment]
My first constructive criticism is you've copied (inadvertently or not) a few rooms from other stations either directly or because if you use the same general shape there's a limited amount of ways you can really do it.  

When this is accepted we'll have to work at refining the concept by bringing more individuality to the component parts of the whole.  Otherwise it's going to be Box/Meta/Deff but Different.

**Comment by brndd** [comment]
Here's the vox trading post and the combined mining+research outpost along with the roid

full

**Comment by Kurfursten** [comment]
General feedback, and I'm sure there will be more quibbles to come after testing - but the chapel looks a bit large. Truth is, you want a manageable size chapel to keep construction on gimmicks smallish. Box is actually about twice what it needs to be (packed is the closest to perfect but it undershoots). For an example, the other day someone got killed in the Box chapel *while a sec officer was in the same chapel* and the officer never noticed. That's big. Here, the same thing could happen *from one side of the cross to the other... both horizontally and vertically*. Who's going to sit in the back pews, almost 2 full screens away from the 'preaching' area? Bring the office in to the carpet intersection and crunch the whole preaching zone down to a manageable size.

**Comment by bungofungo** [comment]
The bar looks nice, it has my blessings.

**Comment by brndd** [comment]
@Probe1 

>My first constructive criticism is you've copied (inadvertently or not) a few rooms from other stations either directly or because if you use the same general shape there's a limited amount of ways you can really do it.

It's a bit of both. I avoided literally copypasting rooms for mapping hygiene reasons (that's how you get retarded varedits), but several rooms are either inspired by or straight up ripped off of other stations. Can't think of any examples off the top of my head, though. It is also true that there's not that many ways you can lay out many of the offices.

That said, I think it's perfectly fine for a station to be Box/Meta/Deff but different. Not every station needs to be completely unique. The main thing that sets this one apart from the others is the segmented design and the use of \~negative space\~ (read: space gaps) for looks. There are other minor innovations or attempts at thereof; for instance, when designing the central section I tried putting opportunities for player interaction there so that when you run through it, you'll have a good chance of witnessing something interesting happening.

I also tried designing the maintenance in a way that gives you alternate routes and shortcuts between sections. In the end the maint didn't end up being quite as grand as I'd envisioned it, and some areas of it are still quite barren (I find cluttering maint incredibly draining), but I don't think it's the worst thing ever.


@Kurfursten 

>the chapel looks a bit large.

The gameplay reason for it is to give the chaplain space to build. I always found the Box chapel far to small to properly build shit in.

The other reason for it is that I saw that big cross-shaped carpet thing on one of the Goon maps and thought it looked dope and wanted it on my map.

File: maps/wheelstation.dm (hunk 3 above)
Comment (brndd) [reply maps/wheelstation.dm]: Deff does the same thing, though I couldn't help but shitpost in the comment there.

**Comment by N3X15** [comment]
OK, one big potential gripe I see is that you've blown up telescience.  

Another issue is the upload being right next to the brig bathrooms. Like, just take out a wall and you can fuck with the consoles, or steal them, and since you're in the bathrooms, nobody'd see you.

The chapel is fucking enormous and need to be less than half the size it is now.

Dragging the defendant in a court case through the peanut gallery is a Bad Idea.

Need windows into holodeck from the hallway.

Morgue is tiny for the map.

Might want to set up some canister pads for topping off xeno for when they inevitably run out of air.

Where the fuck is the bomb testing range?

**Comment by jknpj** [comment]
> Where the fuck is the bomb testing range?

Next to toxins, in the roid.

**Comment by ShiftyRail** [comment]
>//fuck greyshirts and fuck powergamers

Random wires do fuck all for greyshirts and powergamers ; they already have insulated gloves, so they can just cut down every wire starting from the bottom until one clicks

It does hurt people who want to fix wires (engineers), people trapped in a room and need to get out fast, paramedics who lack proper access to save people, and last but not least, it makes dealing with rogue silicons an order of magnitude more difficult

This change do the exact opposite of what you want it to do

edit; I'm wrong

**Comment by N3X15** [comment]
TEG burnchamber doors are science doors.

Get rid of most of those rad collectors around the singularity engine.  Engineering should not have endgame power that soon into the round. Remember to tear up the wiring for them, too.  If they want the jiggawatts, they need to work for them.

Recycling room railgun is firing into a wall.

Chapel railgun is firing into a wall.

**Comment by brndd** [comment]
@ShiftyRail 

The randomized wires are basically the same as on Defficiency, applying only to sec+bridge doors. I think I extended it to vault doors -- for reasons that are rather obvious, I should hope. I don't know why this is a map-specific thing that we're doing rather than global, though.

edit: defficiency apparently also doesn't have this on bridge (command) doors while I do. I think that also makes sense, though.

@N3X15 

The mass drivers are misaligned due to a map rendering bug. Basically all missing pipes are also map rendering bugs (except the ones that aren't, thanks Bomber Harris on Discord). The TEG burn chamber doors are science doors because they're science doors on Boxstation. I tried switching them into engineering doors and it just didn't look right... but maybe that's just me.

**Comment by ShiftyRail** [comment]
Brig doesn't seem to have a convenient way to interrogate prisoners, which will lead to same "problem" metastation and deff have, namely that the prisoner processing will take place in the lobby
Not a major issue, but it does tend to leave the lobby filled with random trash, and I like to have a designated processing room

The brig cells seem to be too big for the flasher to work properly, if the prisoner stands next to the window, he will be out of range

Half of the cells are out of public view, which means they are much less likely to be used (for evidence of this, compare deff or bagel's cells usage to, say, Box or Packed)

Would like to have a direct access from the courtroom to the brig, rather than going through the lobby, but it's a detail

>I don't know why this is a map-specific thing that we're doing rather than global, though.

I personally think it's a very silly feature, using an OOC logic (fuck "powergamers") to break IC logic (doors and circuitry ought to be standardised in a space station), but I can see the rationale for it in security and bridge (even though I still believe it's very dumb)

**Comment by Dacendeth** [comment]
there's no metal and glass stack in the spacepod area.

**Comment by Lunstra** [comment]
The way you've described the atmos setup is weird, it helps mitigate the impact of atmos sabotage from outside engineering which in my experience was always the least common and least damaging atmos sabotage you could do, while actual in house sabotage is exactly as easy as before since all the loops are fed from a single mix pipe anyway. 'bout the only real benefit is that it stops a single breach in the station instantly giving distro the SUCC, instead only emptying that sections distro, which to be fair is a mighty fine benefit.

**Comment by brndd** [comment]
@ShiftyRail 

>Brig doesn't seem to have a convenient way to interrogate prisoners, which will lead to same "problem" metastation and deff have, namely that the prisoner processing will take place in the lobby
Not a major issue, but it does tend to leave the lobby filled with random trash, and I like to have a designated processing room

The interrogation room is past the cells. Maybe it's a little out of the way, but surely sec officers can't be *that* lazy?

>The brig cells seem to be too big for the flasher to work properly, if the prisoner stands next to the window, he will be out of range

I've tested them and they seem to work no matter where you stand in the cell.

@Dacendeth 

>there's no metal and glass stack in the spacepod area.

Map rendering bug.

**Comment by ShiftyRail** [comment]
>The interrogation room is past the cells. Maybe it's a little out of the way, but surely sec officers can't be that lazy?

Not really a matter of laziness but rather of having always something to do
Alternatively, you kinda want the interrogation to happen in view of other officers or the warden so that they can be updated on what's happening

The processing room in Deff isn't even as far away as this one is, and I barely see it used (because everyone buckles prisoners in the "foyer"/meeting room) 
Just talking out of experience here

**Comment by PJB3005** [comment]
I can personally attest to not even realizing there was a prisoner processing room last time I was playing deff.

Granted, we had 4 people to process which wouldn't have fit anyways.

**Comment by IzayaChan** [comment]
Is bagel hidden somewhere?

**Comment by ShiftyRail** [comment]
The observation room next to the interrogation room is way too small and claustrophobic to ever be used by anyone, and the little angle the walls make looks weird to me

**Comment by Gazathor** [comment]
but mape ez no wheel, y u call wheel whn iz no weel?

**Comment by brndd** [comment]
@ShiftyRail 

>The observation room next to the interrogation room is way too small and claustrophobic to ever be used by anyone, and the little angle the walls make looks weird to me

Other maps like box have that room as 4x2 and it's still practically never used by anyone. Might as well conserve the space while still leaving it there for that 0.01% chance that someone will actually want to use it.

**Comment by ShiftyRail** [comment]
Disagree on that observation
The main idea behind this room is that people can observe the interview together (IMO a lot of funny situations in spess come from watching something happen with other spacemen in a line)

This room doesn't allow that

**Comment by brndd** [comment]
@Lunstra 

>The way you've described the atmos setup is weird, it helps mitigate the impact of atmos sabotage from outside engineering which in my experience was always the least common and least damaging atmos sabotage you could do, while actual in house sabotage is exactly as easy as before since all the loops are fed from a single mix pipe anyway. 'bout the only real benefit is that it stops a single breach in the station instantly giving distro the SUCC, instead only emptying that sections distro, which to be fair is a mighty fine benefit.

I don't know if hooking distro to waste is a common thing in practice, but it's one thing that I've wanted to fix about atmos. It's very simple to do, can completely cripple the atmos system, and can be borderline impossible to fix because of how difficult it is to pinpoint where it happened. I thought about different ways to fix it, and fixing it by mapping seemed like the most practical one, so that's something I wanted to try on this map.

The other reason I wanted it was targeted plasmafloods and other gimmicks this setup may enable. You could argue that there's never an objective reason to only plasmaflood one section instead of a specific one, especially given that it requires extra piping effort on the default atmos setup. To that I would respond that this isn't the kind of game where you always do what's objectively best, I guess. The intention was never to make atmos sabotage from within atmos harder; if anything I wanted to provide *more* tools for doing that.

The benefit you mention (one section getting a breach not fucking over the distro of all the other sections) is a happy accident, and to tell you the truth I hadn't really even thought of that.

File: maps/wheelstation.dm (hunk 3 above)
Comment (brndd) [reply maps/wheelstation.dm]: I removed these because I intend to make a separate global PR pertaining to this at some point in the future.

**Comment by brndd** [comment]
Alright I've axed the randomized wires.

File: maps/_map_override.dm [hunk 4]
```diff
@@ -42,5 +42,9 @@
 		#undef MAP_OVERRIDE
 		#include "tgstation.dm"
 		#define MAP_OVERRIDE 7
+	#elif MAP_OVERRIDE == 8
```
Comment (DamianX) [inline maps/_map_override.dm:45]: can you also push the related entry in __compile_options?

Comment (brndd) [reply maps/_map_override.dm:45]: Sure.

File: code/game/machinery/alarm.dm (hunk 1 above)
Comment (brndd) [reply code/game/machinery/alarm.dm]: The air alarm will throw a fit if the pressure in the room drops too low without this preset.

File: __DEFINES/__compile_options.dm [hunk 5]
```diff
@@ -28,6 +28,8 @@
 //#define MAP_OVERRIDE 6
 // tgstation.dm:
 //#define MAP_OVERRIDE 7
+// weheelstation.dm
```
Comment (N3X15) [inline __DEFINES/__compile_options.dm]: Typo

Comment (brndd) [reply __DEFINES/__compile_options.dm]: curses

**Comment by robosperg** [comment]
lots of fire alarms missing for rooms
also dont see any firelocks but not sure if just not rendered in picture

**Comment by N3X15** [comment]
What are you using to generate this map image because it's kinda awful

**Comment by brndd** [comment]
The in-game map rendering verb. It's to my knowledge the only tool that works for our codebase. SpacemanDMM doesn't work, at least that's what @Probe1 told me.

**Comment by brndd** [comment]
@robosperg 

>lots of fire alarms missing for rooms

I tried to be careful with these, but there's always some you miss. Not much that can be done about that except for adding them as they're spotted. I've added one in the bridge's meeting room now, as I noticed it was missing.

> also dont see any firelocks but not sure if just not rendered in picture

I mostly have them under doors. I didn't want to use them to divide hallways *too* much because they're annoying as hell when overused and don't actually help all that much (people just crowbar them open as they walk through the hallway which defeats their purpose). There's probably fewer of them in the hallways than on other maps, which may or may not be a good thing.

**Comment by Durgen** [comment]
>internal trader dock

based

**Comment by Durgen** [comment]
my only suggestion is more maint tumors

edit: also a smaller chapel would have more utility

**Comment by brndd** [comment]
I've updated the bar because a guy in the thread convinced me that wood bars are lame. It's now got the oldschool bar tiling.



I've also updated the full render in the OP of this PR with whatever minor changes have been made since.

**Comment by Durgen** [comment]
as someone who plays a lot of atmos tech, i really like the distro gimmick

File: code/game/machinery/computer/camera.dm (hunk 2 above)
Comment (brndd) [reply code/game/machinery/computer/camera.dm]: Fair enough I'll split it off.

**Comment by brndd** [comment]
I like the chapel personally, though it could do with a bit more clutter or something. I want to see what people do with it before reworking it to be smaller. It's not like its size is really in the way of anything important.

**Comment by Lunstra** [comment]
There is absolutely nothing you can do with that chapel to unfuck it beyond shrinking it, look at it man its about the same size as the entire bar and kitchen area. One of the most common complaints about bagel was how fucking HUGE the chapel was and that it wasted space. Shrinking it is probably your best option, maybe you could repurpose some of the space into a graveyard with some roid blocks or something? The chaplain office is similarly fucking HUGE like holy shit its nearly 3x the size of the CMO office

**Comment by brndd** [comment]
I understand the concern to a degree, but I don't consider it a blocker. Its size is a self-contained issue in that it shouldn't affect the rest of the station. It would be a more pressing concern if the chapel's size created a very long and boring hallway or something.

One of the reasons the chapel grew so large was that I wanted it to have a good view into space, which proved tricky given the constraints of the surroundings. A future rework of it would possibly see swapping it around with the holodeck, but I foresee some layout compromises there that I'm not exactly happy with either; the chapels doors would no longer be at the end of its hallway, which makes it less cool, and perhaps more crucially, the holodeck is shit and needs to have the doors facing east and west at the exact spacing they are now (in addition to having to be the exact dimensions it is now).

So in short, the shape of the chapel is unlikely to change before I can see how it actually works in practice. The furnishings might change though. I'm itching to add a sofa in there somewhere. Maybe replace the back pews with sofas.

**Comment by Kurfursten** [comment]
Escapechapel with a commanding 240 degree view of space

image

**Comment by PJB3005** [comment]
That chapel really wouldn't work, getting maintenance connected to that just isn't feasible with the current layout.

**Comment by brndd** [comment]
Yeah and the holodeck would form a long awkward hallway (even longer and awkwarder than the currenty hallway to it) to reach its doors. It's possible that chapel spaceview is a thing that has to be sacrificed if it is to be made substantially smaller.

File: code/game/machinery/computer/camera.dm [hunk 6]
```diff
@@ -96,7 +96,6 @@ var/list/obj/machinery/camera/cyborg_cams = list(
 	network = list(CAMERANET_THUNDER)
 	density = 0
 	circuit = null
-	layer = ABOVE_WINDOW_LAYER
```
Comment (Kurfursten) [inline code/game/machinery/computer/camera.dm]: ?

Comment (brndd) [reply code/game/machinery/computer/camera.dm]: That's not in the diff anymore so far as I can tell.

Comment (brndd) [reply code/game/machinery/computer/camera.dm]: Though it could be a result of splitting this off into a separate PR. I'll merge Bleeding-Edge into this branch, I guess.

**Comment by brndd** [comment]
>Dragging the defendant in a court case through the peanut gallery is a Bad Idea.

I've rotated the courtroom to address this.

**Comment by AI-Is-Rogue** [comment]
>maint is spacious but isn't bloated with weird-ass junk that shouldn't be there

It almost gives me the feeling that it's an actual maintenance tunnel, you know, as opposed to somehow stumbling upon a fully stocked circus grounds when you go out back to fix a window on your state-of-the-art space station. big 👍 on that, only deff should have bloated maint tumors

**Comment by DiscountDirtyDan** [comment]
I actually quite like this map. Despite some of the flaws pointed out I think this could be a worthy replacement for Deff as the high pop map once said flaws are fixed. The biggest imo is the randomized wires. Fuck that noise. Most people learn the wires by using another door before they hack into the bridge to rush the spare or some shit so it honestly serves no other purpose than to over-complicate things for those who do their jobs while doing nothing to stop those who refuse to.

**Comment by brndd** [comment]
There are no randomized wires on this map anymore. Though I should say they would prevent that specific scenario you mentioned (learning the wires on another door and then hacking into the bridge).

**Comment by Durgen** [comment]
a longer cargo shuttle outflow belt would speed up unloading large shuttles

**Comment by incakola** [comment]
Maybe this is a great station in live play.

The distro loops thing is a nice idea.
Hallways zig zag a bit.
Maint has a bit of zig zag also.
Some maint doesn't really lead anywhere (eg EVA maint), but maybe this is really good for hiding given there are very few maint tumors?
Some rooms might be too big. 
There are blindspots like Bar, Hydro, Chapel, Chemistry/MedbayLobby, Xenobio, MedbayCryoSleeperArea.
Atmos, Cargo, Singularity could be a tile or two smaller.
Singularity has so many roundstart rad collectors?
Is it a good idea to have the AI, Bridge, Vault, and Sec so close?

We could just play it and find out then let the feedback decide what needs attention.

**Comment by brndd** [comment]
It just hit me that I'd forgotten to put in the gateway -- clearly a room of great importance, since nobody else seemed to notice its absence either, heh.

Anyway, there was a convenient gap near the bridge that I could snugly fit it in. It's not the grandest gateway ever, but I daresay this might be one of the least used features in the game so dedicating more space to it would feel a little silly.

**Comment by VonDudeMaster** [comment]
Let the players expand the collector array around the singularity on their own. Otherwise you're taking the fun away from this engine.
If you want your map to maximize the singulo, expand the area around it like you already did and then a little bit more. Or just leave the area around it ready for expansion so the players don't need to remove a dozen reinforced walls just to get there.
Singulo Field

Very nice work by the way.

**Comment by brndd** [comment]
That looks like a very reasonable design tweak.

The reason I put in so many rad collectors was that I wanted to create a roundstart singulo that makes at least a little sense to set up over the much safer and much more powerful SME and TEG. But your point that it being all set up like this takes away from the fun of engineering is fair.

I'm axing all but the default 6 rad collectors from around the singulo and extending the arms to run almost a full circle around it. The reason I want to keep the funky encircled design is that I want this map to provide a reasonable opportunity to set up a truly powerful singulo without massive construction effort. I guess it might also be a reasonably good location to set up the RUST in.

I think I'd also like to make the SME a bit more unbuilt. If you happen to have any feedback regarding that, please let me know. Otherwise I'm probably just going to imitate the Box SME by setting up an unfinished suck-blow based cooling setup.

edit: here's how it looks like now

**Comment by VonDudeMaster** [comment]
That singulo would be perfect if not for a tiny problem I forgot to mention.

Expanding the singularity implies installing 26 more collectors. That's 26 plasma tanks that need to be filled inside that massive room.
There needs to be some kind of compartmentalization preventing the whole engine room from being accidentally flooded with plasma and burned. A window+airlock+window leading south on both sides, for example.

As for removing stuff from the SME, start with the HE loop inside the chamber, the volume pump, and at least one tank storage unit. Any more than that and it will start to look like deff's SME with only six collectors. So leave all 16 in and unwrench them.
Or maybe just leave some engineering crates with collectors and emitters in them and let the players figure it out?

**Comment by brndd** [comment]
I added some windoors in there as an unobtrusive way of dividing the room. Dividing it is a good catch, because without them you can accidentally open a canister at the far end of the singularity room and it'll fill all of engineering with plasma. I also added some airlocks in the hallways above the particle accelerator to rectify this problem.

As for the SME, I settled with removing some of the HE pipes inside the chamber and moving the radiation collectors into crates in engineering storage. Putting them in crates makes it more convenient to stick them up around the singulo anyway, as otherwise you'd have to unwrench the ones in the SME room and move those.

There's an updated map render in the OP that shows these changes.

**Comment by brndd** [comment]
I've expanded the interrogation room observation as @ShiftyRail suggested. Hopefully it's a bit less claustrophobic now.
